src/rules/extensions.js

```diff
@@ -147,24 +147,46 @@ module.exports = {
         const extensionForbidden = isUseOfExtensionForbidden(extension)
         if (extensionRequired && !extensionForbidden) {
           context.report({
-            node: source,
+            node,
             message:
               `Missing file extension ${extension ? `"${extension}" ` : ''}for "${importPath}"`,
           })
         }
       } else if (extension) {
         if (isUseOfExtensionForbidden(extension) && isResolvableWithoutExtension(importPath)) {
           context.report({
-            node: source,
+            node,
             message: `Unexpected use of file extension "${extension}" for "${importPath}"`,
           })
         }
       }
     }
 
+    function checkImportFileExtension(node) {
+      const { source } = node
+
+      // bail if the declaration doesn't have a source, e.g. "export { foo };"
+      if (!source) return
+
+      const importPath = source.value
+
+      checkFileExtension(importPath, source)
+    }
+
+    function checkCommonJSFileExtension(node) {
+      if (!props.commonjs) return
+
+      if (!isStaticRequire(node)) return
+
+      const importPath = node.arguments[0].value
+
+      checkFileExtension(importPath, node)
+    }
+
     return {
-      ImportDeclaration: checkFileExtension,
-      ExportNamedDeclaration: checkFileExtension,
+      ImportDeclaration: checkImportFileExtension,
+      ExportNamedDeclaration: checkImportFileExtension,
+      CallExpression: checkCommonJSFileExtension,
```

Comment:
maybe we could set this to null if props.commonjs is falsy, and avoid the visitor entirely?

Reply:
Will make the change.

Reply:
Rather than setting it to `null` (this generated some errors on CI) or use `undefined` in a ternary operator, the assignment is wrapped in an `if` statement.

Reply:
aw, i'd actually prefer the inline `undefined` to mutating the object.